### PR TITLE
BAEL-3830: Test a mock JNDI datasource with Spring

### PR DESCRIPTION
**Pull request by azhwani**
Simple PR for "Test a mock JNDI datasource with Spring" article

### CODE REVIEW

File: persistence-modules/spring-persistence-simple-2/pom.xml [hunk 1]
```diff
@@ -53,6 +60,8 @@
         <org.springframework.version>5.2.4.RELEASE</org.springframework.version>
         <!-- persistence -->
         <h2.version>1.4.200</h2.version>
+        <!-- simple-jndi  -->
```
Comment (JonCook) [inline persistence-modules/spring-persistence-simple-2/pom.xml:63]: There is an extra space in the comment text here.

File: persistence-modules/spring-persistence-simple-2/pom.xml [hunk 2]
```diff
@@ -32,6 +32,13 @@
             <version>${h2.version}</version>
             <scope>test</scope>
         </dependency>
+
```
Comment (JonCook) [inline persistence-modules/spring-persistence-simple-2/pom.xml:35]: I looks like we have got some tabs in the pom.xml here.

File: persistence-modules/spring-persistence-simple-2/src/test/java/com/baeldung/jndi/datasource/SimpleJNDIUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class SimpleJNDIUnitTest {
+
+    private InitialContext initContext;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        System.setProperty("java.naming.factory.initial", "org.osjava.sj.SimpleContextFactory");
```
Comment (JonCook) [inline persistence-modules/spring-persistence-simple-2/src/test/java/com/baeldung/jndi/datasource/SimpleJNDIUnitTest.java:18]: If we are going to use System properties shouldn't we reset them when the test tears down? We don't want any unexpected side effects when other tests run.

File: persistence-modules/spring-persistence-simple-2/pom.xml (hunk 1 above)
Comment (azhwani) [reply persistence-modules/spring-persistence-simple-2/pom.xml]: Done!

File: persistence-modules/spring-persistence-simple-2/src/test/java/com/baeldung/jndi/datasource/SimpleJNDIUnitTest.java (hunk 3 above)
Comment (azhwani) [reply persistence-modules/spring-persistence-simple-2/src/test/java/com/baeldung/jndi/datasource/SimpleJNDIUnitTest.java:19]: Done!

File: persistence-modules/spring-persistence-simple-2/pom.xml (hunk 2 above)
Comment (azhwani) [reply persistence-modules/spring-persistence-simple-2/pom.xml]: Done!

**Pull request merged**